modules/dnn/src/layers/cpu_kernels/softmax.hpp

```diff
@@ -0,0 +1,51 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+// This file is modified from the ficus (https://github.com/vpisarev/ficus/blob/master/lib/NN/OpNN.fx).
+// Here is the original license:
+/*
+    This file is a part of ficus language project.
+    See ficus/LICENSE for the licensing terms
+*/
+
+#ifndef OPENCV_DNN_SOFTMAX_HPP
+#define OPENCV_DNN_SOFTMAX_HPP
+
+#include "opencv2/core/hal/intrin.hpp"
+#include <opencv2/dnn/shape_utils.hpp>
+
+namespace cv { namespace dnn {
+
+struct SoftmaxOpt {
+    bool use_avx;
+    bool use_avx2;
+    bool use_neon;
+    bool use_lasx;
+
+    SoftmaxOpt() {
+        use_avx = false;
+        use_avx2 = false;
+        use_neon = false;
+        use_lasx = false;
+    }
+
+    void init() {
+        use_avx = checkHardwareSupport(CPU_AVX);
+        use_avx2 = checkHardwareSupport(CPU_AVX2);
+        use_neon = checkHardwareSupport(CPU_NEON);
+        use_lasx = checkHardwareSupport(CPU_LASX);
+    }
+
+    bool all() {
+        return use_avx || use_avx2 || use_neon || use_lasx;
+    }
+};
```

Comment:
This is the exact same with `GemmOpt`,

https://github.com/opencv/opencv/blob/617d7ff575200c0a647cc615b86003f10b64587b/modules/dnn/src/layers/cpu_kernels/fast_gemm.hpp#L20-L43

I think we should not make duplicates here. It's better to make a general one and use it everywhere.

Reply:
Should this Opt belong to all layers or just for cpu kernel? Should I change it for `convolution.hpp`

Reply:
We can keep it as-is, then submit a PR to change this after merge.